frontend/src/tests/application-tests.js

```diff
@@ -104,41 +233,294 @@ export function applicationsTests({ apiGET, apiPOST }) {
+                previous_university_ta: submittedData.previous_university_ta,
+                previous_experience_summary:
+                    submittedData.previous_experience_summary,
+                gpa: submittedData.gpa,
+                comments: submittedData.comments,
+            };
+
+            await switchToUser(userCreatedFromApplicant);
+
+            // Read survey.js posting data
+            resp = await apiGET(
+                `/public/postings/${postingData.url_token}`,
+                true
+            );
+            expect(resp).toHaveStatus("success");
+            checkPropTypes(surveyPropTypes, resp.payload);
+            expect(resp.payload["prefilled_data"]).toEqual(
+                expect.objectContaining(applicationPrefillData)
+            );
+            await restoreDefaultUser();
+        });
+
+        it.skip(
+            "When submitting survey.js data cannot add a position_preference for a position not listed in the" +
+                " posting",
+            async () => {
+                // Submit a position preference for a position_code that isn't
+                // associated with the current application
+                // See issue #621
+                await switchToUser(userCreatedFromApplicant);
+
+                // PF ==> Position Preference
+                const applicationExistentPF = {
+                    answers: {
+                        ...applicant,
+                        position_preferences: {
+                            // position that already exists but isn't associated
+                            // with the current application
+                            [position.position_code]: HIGH_PREFERENCE,
+                        },
+                    },
+                };
+
+                let resp = await apiPOST(
+                    `/public/postings/${postingDataForPFTests.url_token}/submit`,
+                    applicationExistentPF,
+                    true
+                );
+                expect(resp).toHaveStatus("error");
+
+                const applicationNonExistentPF = {
+                    answers: {
+                        ...applicant,
+                        position_preferences: {
+                            // position that already exists but isn't associated
+                            // with the current application
+                            DNE101: HIGH_PREFERENCE,
+                        },
+                    },
+                };
+
+                resp = await apiPOST(
+                    `/public/postings/${postingDataForPFTests.url_token}/submit`,
+                    applicationNonExistentPF,
+                    true
+                );
+                expect(resp).toHaveStatus("error");
+
+                await restoreDefaultUser();
+            }
+        );
+
+        it("When submitting survey.js data in an incorrect matter, submission fails with error", async () => {
```

Comment:
Not sure if `When submitting survey.js data in an incorrect matter` makes clear sense.